odo application describe crash when run out of a component folder

/kind bug


## What versions of software are you using?

**Operating System:**
Win10Pro

**Output of `odo version`:**
```
odo v1.0.3 (HEAD)

Server: https://172.18.1.78:8443
Kubernetes: v1.11.0+d4cacc0
```

## How did you run odo exactly?

`odo application describe myapp --project myproj`

## Actual behavior
```
Application Name: sbapp has 1 component(s) and 1 service(s):                                                                                               
--------------------------------------                                                                                                                     
Component Name: sbcomp1                                                                                                                                    
Type: java                                                                                                                                                 
Source: file:///./                                                                                                                                         
Environment Variables:                                                                                                                                     
 · DEBUG_PORT=5858                                                                                                                                         
panic: runtime error: slice bounds out of range                                                                                                            
                                                                                                                                                           
goroutine 1 [running]:                                                                                                                                     
github.com/openshift/odo/pkg/odo/util.PrintComponentInfo(0xc00095edc0, 0xc00057bc58, 0x7, 0x19d4618, 0x9, 0x19e9cb5, 0x19, 0xc00057bc58, 0x7, 0x0, ...)    
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/util/cmdutils.go:116 +0x13b7                                                     
github.com/openshift/odo/pkg/odo/cli/application.(*DescribeOptions).Run(0xc00084f230, 0x0, 0x0)                                                            
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/application/describe.go:88 +0x4ec                                            
github.com/openshift/odo/pkg/odo/genericclioptions.GenericRun(0x1ca18e0, 0xc00084f230, 0xc0007e4480, 0xc000990030, 0x1, 0x3)                               
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/genericclioptions/runnable.go:28 +0x107                                          
github.com/openshift/odo/pkg/odo/cli/application.NewCmdDescribe.func1(0xc0007e4480, 0xc000990030, 0x1, 0x3)                                                
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/application/describe.go:117 +0x65                                            
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).execute(0xc0007e4480, 0xc000990000, 0x3, 0x3, 0xc0007e4480, 0xc000990000)                
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:702 +0x290                                      
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0007ed200, 0x1ad86e0, 0xc000644d80, 0x0)                                     
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:783 +0x2d1                                      
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).Execute(...)                                                                             
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:736                                             
main.main()                                                                                                                                                
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/cmd/odo/odo.go:72 +0x3f6                                                                 
```

## Expected behavior

Should not crash

## Any logs, error output, etc?

--- a/pkg/odo/cli/application/describe.go
+++ b/pkg/odo/cli/application/describe.go
@@ -85,7 +85,7 @@ func (o *DescribeOptions) Run() (err error) {
 				for _, currentComponent := range componentList.Items {
 					componentDesc, err := component.GetComponent(o.Client, currentComponent.Name, o.appName, o.Project)
 					util.LogErrorAndExit(err, "")
-					util.PrintComponentInfo(o.Client, currentComponent.Name, componentDesc, o.Application, o.Project)
+					util.PrintComponentInfo(o.Client, currentComponent.Name, componentDesc, o.appName, o.Project)
 					fmt.Println("--------------------------------------")
 				}
 			}

--- a/pkg/odo/util/cmdutils.go
+++ b/pkg/odo/util/cmdutils.go
@@ -7,10 +7,10 @@ import (
 	"unicode"
 
 	"github.com/openshift/odo/pkg/component"
-	"github.com/openshift/odo/pkg/config"
 	"github.com/openshift/odo/pkg/log"
 	"github.com/openshift/odo/pkg/machineoutput"
 	"github.com/openshift/odo/pkg/occlient"
+	"github.com/openshift/odo/pkg/storage"
 	urlPkg "github.com/openshift/odo/pkg/url"
 
 	"github.com/golang/glog"
@@ -72,10 +72,6 @@ func CheckOutputFlag(outputFlag string) error {
 
 // PrintComponentInfo prints Component Information like path, URL & storage
 func PrintComponentInfo(client *occlient.Client, currentComponentName string, componentDesc component.Component, applicationName string, project string) {
-	localConfig, err := config.New()
-	if err != nil {
-		LogErrorAndExit(err, "")
-	}
 
 	log.Describef("Component Name: ", currentComponentName)
 	log.Describef("Type: ", componentDesc.Spec.Type)
@@ -95,26 +91,32 @@ func PrintComponentInfo(client *occlient.Client, currentComponentName string, co
 		}
 
 		// Cut off the last newline and output
-		output = output[:len(output)-1]
-		log.Describef("Environment Variables:\n", output)
+		if len(output) > 0 {
+			output = output[:len(output)-1]
+			log.Describef("Environment Variables:\n", output)
+		}
+
 	}
 
 	// Storage
 	if len(componentDesc.Spec.Storage) > 0 {
 
 		// Retrieve the storage list
-		storages, err := localConfig.StorageList()
+		storages, err := storage.List(client, currentComponentName, applicationName)
 		LogErrorAndExit(err, "")
 
 		// Gather the output
 		var output string
-		for _, store := range storages {
-			output += fmt.Sprintf(" · %v of size %v mounted to %v\n", store.Name, store.Size, store.Path)
+		for _, store := range storages.Items {
+			output += fmt.Sprintf(" · %v of size %v mounted to %v\n", store.Name, store.Spec.Size, store.Status)
 		}
 
 		// Cut off the last newline and output
-		output = output[:len(output)-1]
-		log.Describef("Storage:\n", output)
+		if len(output) > 0 {
+			output = output[:len(output)-1]
+			log.Describef("Storage:\n", output)
+		}
+
 	}
 
 	// URL
@@ -138,8 +140,11 @@ func PrintComponentInfo(client *occlient.Client, currentComponentName string, co
 			}
 		}
 		// Cut off the last newline and output
-		output = output[:len(output)-1]
-		log.Describef("URLs:\n", output)
+		if len(output) > 0 {
+			output = output[:len(output)-1]
+			log.Describef("URLs:\n", output)
+		}
+
 	}
 
 	// Linked components
@@ -156,8 +161,10 @@ func PrintComponentInfo(client *occlient.Client, currentComponentName string, co
 		}
 
 		// Cut off the last newline and output
-		output = output[:len(output)-1]
-		log.Describef("Linked Components:\n", output)
+		if len(output) > 0 {
+			output = output[:len(output)-1]
+			log.Describef("Linked Components:\n", output)
+		}
 
 	}
 
@@ -178,18 +185,25 @@ func PrintComponentInfo(client *occlient.Client, currentComponentName string, co
 				for i := range secrets.Data {
 					secretOutput += fmt.Sprintf("    · %v\n", i)
 				}
-				// Cut off the last newline
-				secretOutput = secretOutput[:len(secretOutput)-1]
-				output += fmt.Sprintf(" · %s\n   Environment Variables:\n%s\n", linkedService, secretOutput)
+
+				if len(secretOutput) > 0 {
+					// Cut off the last newline
+					secretOutput = secretOutput[:len(secretOutput)-1]
+					output += fmt.Sprintf(" · %s\n   Environment Variables:\n%s\n", linkedService, secretOutput)
+				}
+
 			} else {
 				output += fmt.Sprintf(" · %s\n", linkedService)
 			}
 
 		}
 
-		// Cut off the last newline and output
-		output = output[:len(output)-1]
-		log.Describef("Linked Services:\n", output)
+		if len(output) > 0 {
+			// Cut off the last newline and output
+			output = output[:len(output)-1]
+			log.Describef("Linked Services:\n", output)
+
+		}
 
 	}
 

--- a/tests/integration/cmd_app_test.go
+++ b/tests/integration/cmd_app_test.go
@@ -19,6 +19,8 @@ var _ = Describe("odo app command tests", func() {
 
 	appName := "app"
 	cmpName := "nodejs"
+	mountPath := "/data"
+	size := "1Gi"
 
 	// This is run after every Spec (It)
 	var _ = BeforeEach(func() {
@@ -128,5 +130,18 @@ var _ = Describe("odo app command tests", func() {
 
 			helper.CmdShouldPass("odo", "app", "delete", appName, "--project", project, "-f")
 		})
+
+	})
+
+	Context("When running app describe with storage added in component in directory that doesn't contain .odo config directory", func() {
+		It("should successfully execute describe", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--app", appName, "--project", project, "--context", context)
+			helper.CmdShouldPass("odo", "storage", "create", "storage-one", "--context", context, "--path", mountPath, "--size", size)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			helper.CmdShouldPass("odo", "app", "describe", appName, "--project", project)
+
+		})
+
 	})
 })

